ow_faults/include/ow_faults/FaultDetector.h

```diff
@@ -0,0 +1,47 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef FaultDetector_h
+#define FaultDetector_h
+
+#include <bitset>
+#include <ctime>
+#include <cstdlib>
+#include <ros/ros.h>
+#include <cstdint>
+#include <std_msgs/Float64.h>
+#include <std_msgs/Empty.h>
+#include <ow_faults/FaultsConfig.h>
+#include "ow_faults/SystemFaults.h"
+#include "ow_faults/ArmFaults.h"
+#include "ow_faults/PowerFaults.h"
+#include "ow_faults/PTFaults.h"
+#include "ow_faults/CamFaults.h"
+// #include <ow_lander/lander_joints.h>
+#include <sensor_msgs/JointState.h>
+#include <geometry_msgs/WrenchStamped.h>
+#include <unordered_map>
+#include <random>
+
+
+// This class injects simple message faults that don't need to be simulated
+// at their source. Modified topics are prefixed with "/faults". This could be
+// accomplished on the original topics, but, for example, in the case of
+// /joint_states, this would require forking and modifying joint_state_publisher.
+// Creating a modified version of the original message is simpler, and it
+// leaves the original message intact in case we want to use it as part of a
+// placeholder fault estimator.
+class FaultDetector
+{
+
+public:
+  FaultDetector(ros::NodeHandle& node_handle);
+  ~FaultDetector(){}
+
```

Comment:
For completeness, define, 'delete', or 'default' the copy constructor and assignment operator.

Reply:
They three functions are not used, I think it is better to remove their declarations

Reply:
destructor, copy constructor, assignment operator.